Program Optimization - Updated Program configuration lesson

0xCipherCoder opened the pull request:
### Problem



### Summary of Changes

- Updated code snippets with the latest anchor version
- Fixed content with updated version changes
- Fixed grammar and styling
- Fixed as per guidelines
- Fixed order of content based on realtime code following tutorial.

Fixes #

Unboxed PRs
Starter - https://github.com/Unboxed-Software/solana-admin-instructions/pull/3
Solution - https://github.com/Unboxed-Software/solana-admin-instructions/pull/4
Challenge - https://github.com/Unboxed-Software/solana-admin-instructions/pull/5

<!-- 
Note: The maintainers of this repo may make editorial changes as needed without creating comments.
Please ensure you have "Allow edits by maintainers" setting on this PR enabled to help speed up the review process. Thanks :)
-->

0xCipherCoder commented:
@mikemaccana Please review this once. Let me know if any suggestions.

mikemaccana commented on content/courses/program-optimization/program-configuration.md:
```diff
@@ -17,43 +17,46 @@ description:
+   your build, allowing the code to adjust values based on the specified build.
+2. **Admin-only** program accounts and instructions that are only accessible by
+   the program's upgrade `authority` for setting and managing configuration
+   values post-deployment.
```
Much clearer than the old version. 👍

mikemaccana commented on content/courses/program-optimization/program-configuration.md:
```diff
@@ -537,58 +547,100 @@ Next, update the `config.ts` test file to create a mint using the generated
+...
+
+const deploy = () => {
+  const deployCmd = `solana program deploy --url localhost -v --program-id $(pwd)/target/deploy/config-keypair.json $(pwd)/target/deploy/config.so`
```
`deploy_command` see naming in contributing.

mikemaccana replied:
I would probably also use node (rather than subshells) for `working_directory`, just so we focus on Solana (a lot of people don't understand subshells, hell most people don't understand unix).

0xCipherCoder replied:
Yes understood. I have now changed the snippet to the node version for the working directory.

0xCipherCoder replied:
Also renamed variable to `deploy_command`.

mikemaccana reviewed (approved):
Very minor, just make sure all the unboxed links are updated and we can merge this.

0xCipherCoder commented:
> Very minor, just make sure all the unboxed links are updated and we can merge this.

Thanks @mikemaccana I have updated all the URLs to solana-developers, changed a code snippet to Node, and resolved comments. Please review and let me know if any additional changes required.

mikemaccana commented:
@0xCipherCoder another win. 💪🏼
